EIPS/eip-6239.md

```diff
@@ -0,0 +1,167 @@
+    1. [EIP-5192](./eip-5192.md)’s `ERC5192` (`0xb45a3c0e`)
+
+### Contract Interface
+
+```Solidity
+
+/**
+ * @title Semantic Soulbound Token
+ * Note: the EIP-165 identifier for this interface is 0xfbafb698
+ */
+interface ISemanticSBT{
+    /**
+     * @dev This emits when minting a Semantic Soulbound Token.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the Semantic Soulbound Token. An RDF statement is the statement made by an RDF triple.
+     */
+    event CreateRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+
+
+    /**
+     * @dev This emits when updating the RDF data of Semantic Soulbound Token. RDF data is a collection of RDF statements that are used to represent information about resources.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the semantic soulbound token. An RDF statement is the statement made by an RDF triple.
+     */
+    event UpdateRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+
+
+/**
+     * @dev This emits when burning or revoking Semantic Soulbound Token.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfStatements The RDF statements for the Semantic Soulbound Token. An RDF statement is the statement made by an RDF triple.
+     */
+    event RemoveRDF (
+        uint256 indexed tokenId,
+        string  rdfStatements
+    );
+
+    /**
+     * @dev Returns the RDF statements of the Semantic Soulbound Token. An RDF statement is the statement made by an RDF triple.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+
+     */
+    function rdfOf(uint256 tokenId) external view returns (string memory);
+
+}
+
+```
+
+ISemanticRDFSchema, an extension of EIP-721 Metadata, is **OPTIONAL** for this standard, it is used to get the Schema URI for the RDF data.
+
+```Solidity
+
+interface ISemanticRDFSchema{
+
+    /**
+     * @notice Get the URI of schema for this contract.
+     * @return The URI of the contract which point to a configuration profile.
+     */
+    function schemaURI() external view returns (string memory);
+}
+
+```
+
+ISemanticSBTUpdate is an extension interface, that is **OPTIONAL** for this standard, used to update the RDF data for the Semantic Soulbound Token.
+
+```Solidity
+
+interface ISemanticSBTUpdate{
+    /**
+     * @notice Update the RDF data for Semantic Soulbound Token. Implementors can assign updaters as needed, for example, the token issuer.
+     * @dev Emits the `UpdateRDF` event.
+     * @param tokenId The identifier for the Semantic Soulbound Token.
+     * @param rdfData RDF data is a collection of RDF statements that are used to represent information about resources.
+     */
+    function updateRDF(uint256 tokenId, RDFData memory rdfData) external;
```

Comment:
`RDFData` is not defined within the specification section. The specification section must contain sufficient information to implement the standard.

Reply:
RDFData allows for custom data formats,developers can define their own data structures as needed. Here we provide a general example of the structure of RDFData, using the string type to define the subject, predicate, and object. In most cases, one token points to one SPO, and the data structure in example 1 can be used. For scenarios where multiple predicates describe the same subject, the data structure in example 2 can be used, defining the predicate and object using an array type.

Example 1: Single SPO scenario
Describing User A as a friend of User B:
:Soul_UserA p:friendOf :Soul_UserB
Here, :Soul_UserA is the subject, p:friendOf is the predicate, and :Soul_UserB is the object.

Example 2: Multiple predicate scenario
Describing User A's profile with two attributes, name and age:
:Soul_UserA p:name "Alice"; p:age 20.
Here, :Soul_UserA is the subject, p:name and p:age are predicates, and "Alice" and 20 are objects.

Reply:
Defined RDFData and give some example data structures.
RDFData is a set of RDF statements that are used to represent social information about wallet addresses in the graph model. An RDF statement is the statement made by an RDF triple in subject-predicate-object format. 
RDFData allows customized data schema, and developers can define data structures according to their needs. Here we provide a generic example of the structure of RDFData, using string types to define subject, predicate, and object.

In most cases, one token points to one RDF triple, the data structure as example 1 is applied. For scenarios where multiple predicates describe the same subject, i.e.multiple triples, the data structure as example 2 is applied, defining the predicate and object using an array type.

Example 1: Single triple scenario
Describing Alice attended Devcon :
:Soul_Alice p:attended :Event_Devcon
In this case, :Soul_Alice is the subject, p:attended is the predicate, and :Event_Devcon is the object.

Example 2: Multiple triples scenario
Describing Alice's profile with two attributes, event and year:
:Soul_Alice p:attended :Event_Denv; p:createdAt :2022. 
In this case, :Soul_Alice is the subject, p:attended p:createdAt are predicates, and :Event_Devcon :2022 are object.

Reply:
I'm concerned that if, hypothetically, I wanted to write a generic dapp that can show RDF data (and its updates), I wouldn't be able to decode these events for any arbitrary token, and I'd need to write a specific implementation for each token to deal with the different `RDFData` encodings.